stream-chat-android-client/src/main/java/io/getstream/chat/android/client/notifications/PushTokenRepository.kt

```diff
@@ -0,0 +1,69 @@
+package io.getstream.chat.android.client.notifications
+
+import android.content.Context
+import android.content.SharedPreferences
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.extensions.getNonNullString
+import io.getstream.chat.android.client.logger.ChatLogger
+import io.getstream.chat.android.client.notifications.handler.ChatNotificationHandler
+
+internal class PushTokenRepository(
```

Comment:
Not the best name since it's not only storing entity. Because it's shared preference storing mechanism is its' internal logic, API interface of this is only `updateTokenIfNecessary`. What about something like `PushTokenUpdateHandler`?

Reply:
Or maybe move `updateTokenIfNecessary()` into `ChatNotifications`?

Reply:
Renamed to `PushTokenUpdateHandler `